src/main/java/org/java_lcw/jil/AWTImage.java

```
@@ -0,0 +1,485 @@
+    switch(type) {
+    case TIFF:
+      return AWTImage.fromJavaImage(TiffFile.open(filename));
+    case PNG:
+    case JPEG:
+      return AWTImage.fromBufferedImage(ImageIO.read(new File(filename)));
+    default:
+      throw new ImageException("Could not determen filetype");
+    }
+  }
+
+  public BufferedImage getBufferedImage() {
+    return bi;
+  }
+
+  @Override
+  public void save(File file) throws IOException, ImageException {
+    ImageType t = Utils.getImageType(file);
+    save(file, t);
+  }
+
+  @Override
+  public void save(String file) throws IOException, ImageException {
+    ImageType t = Utils.getImageType(file);
+    save(file, t);
+  }
+
+  @Override
+  public void save(File file, ImageType type) throws IOException, ImageException {
+    save(file.getAbsolutePath(), type);
+  }
+
+  @Override
+  public void save(String filename, ImageType type) throws IOException, ImageException {
+    switch(type) {
+    case JPEG: {
+      JpegFile.save(filename, this);
+    }break;
+    case PNG: {
+      PngFile.save(filename, this);
+    } break;
+    case TIFF: {
+      TiffFile.save(filename, this);
+    } break;
+    default:
+      break;
+    }
+  }
+
+
+  @Override
+  public JavaImage toJavaImage() {
+    return JavaImage.fromBufferedImage(bi);
+  }
+
+  @Override
+  public AWTImage changeMode(byte MODE) {
+    BufferedImage nbi;
+    if(MODE == Image.MODE_L) {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_BYTE_GRAY);
+    } else if(MODE == Image.MODE_RGB) {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_3BYTE_BGR);
+    } else {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR);
+    }
+    nbi.getGraphics().drawImage(bi, 0, 0, null);
+    return new AWTImage(nbi, MODE);
+  }
+
+  @Override
+  public AWTImage resizeWithBorders(int bWidth, int bHeight, Color borderColor,
+      ScaleType st) {
+    return null;
```

Seems like this should throw an UnsupportedOperationException?